src/anyGPT/service/app.py

```diff
@@ -0,0 +1,63 @@
+import argparse
+import uvicorn
+from fastapi import FastAPI, Depends
+from pydantic import BaseModel
+from anyGPT.inference.runner import AnyGPTRunner
+
+
+# Parsing for cli
+def _create_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        prog="anyGPT inference service",
+        description="Loads an anyGPT model and hosts it on a simple microservice that can run inference over the network.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    parser.add_argument(
+        "model", action="store", help="Path t0 the trained model checkpoint to load"
+    )
+    parser.add_argument(
+        "--port",
+        action="store",
+        type=int,
+        default=5000,
+        help="Port to start the microservice on",
+    )
+    parser.add_argument(
+        "--log-level", action="store", default="info", help="uvicorn log level"
+    )
+
+    return parser
+
+
+# Pydantic DataModel to define request data
+class InferenceRequest(BaseModel):
+    data: str
+
+
+# Main app
+app = FastAPI()
+model = ""
+
+
+# Runner dependency
+def get_runner():
+    return AnyGPTRunner(model)
+
+
+# Runs inference on the model
+@app.post("/infer")
+async def infer(request: InferenceRequest, runner: AnyGPTRunner = Depends(get_runner)):
+    print(f"Running inference on {request.data}")
+    return runner.sample(request.data)
```

Comment:
can you add the additional parameters available to the inputs of the post request? it's max new tokens, temperature, and top k.

Reply:
This is fixed